.github/workflows/self-new-model-pr-caller.yml

```diff
@@ -0,0 +1,124 @@
+name: New model PR CI
+
+on:
+  pull_request:
+    paths:
+      - "src/transformers/models/**"
+
+env:
+  HF_HOME: /mnt/cache
+  TRANSFORMERS_IS_CI: yes
+  OMP_NUM_THREADS: 8
+  MKL_NUM_THREADS: 8
+  RUN_SLOW: yes
+  # For gated repositories, we still need to agree to share information on the Hub repo. page in order to get access.
+  # This token is created under the bot `hf-transformers-bot`.
+  HF_HUB_READ_TOKEN: ${{ secrets.HF_HUB_READ_TOKEN }}
+  SIGOPT_API_TOKEN: ${{ secrets.SIGOPT_API_TOKEN }}
+  TF_FORCE_GPU_ALLOW_GROWTH: true
+  RUN_PT_TF_CROSS_TESTS: 1
+  CUDA_VISIBLE_DEVICES: 0,1
+
+jobs:
+  check_for_new_model:
+      runs-on: ubuntu-22.04
+      name: Check if PR is a new model PR
+      outputs:
+        new_model: ${{ steps.check_new_model.outputs.new_model }}
+      steps:
+        - uses: actions/checkout@v4
+          with:
+            fetch-depth: "0"
+
+        - name: Check if there is new model
+          id: check_new_model
+          run: |
+            python -m pip install GitPython
+            echo "new_model=$(python utils/check_if_new_model_added.py | tail -n 1)" >> $GITHUB_OUTPUT
+
+  run_test:
+      name: Run tests for the new model
+      if: needs.check_for_new_model.outputs.new_model != ''
```

Comment:
```suggestion
      if: needs.check_for_new_model.outputs.new_model != '' && contains(github.actor, 'transformers')
```

Reply:
let's make sure only HF people can trigger this for safety reasons

Reply:
This is great! Nice to have this way to control instead of having to request specific commit messages or running all the time

Reply:
@amyeroberts [My above comment](https://github.com/huggingface/transformers/pull/30341#discussion_r1574831835) only works for outside contributors. For internal contributors, the workflow won't need to be approved to be run - hence every commit will trigger the workflow (for which I think it's not great, but probably acceptable)

Reply:
Mmmm I am not in favor of every comming triggering the workflow. When you add the model you don't want to do this!